sdk/inc/azure/core/internal/az_hfsm_pipeline_internal.h

```diff
@@ -0,0 +1,180 @@
+ *
+ * @param[in] pipeline The #az_hfsm_pipeline to use for this call.
+ * @param[in] event The event being enqueued.
+ * @return An #az_result value indicating the result of the operation.
+ */
+AZ_NODISCARD az_result
+az_hfsm_pipeline_post_outbound_event(az_hfsm_pipeline* pipeline, az_hfsm_event const event);
+
+/**
+ * @brief Sends an inbound event from the current policy.
+ *
+ * @note This function should be used during pipeline processing. The event is processed on the
+ * current thread (stack).
+ *
+ * @param policy The current policy trying to send the event.
+ * @param event The event being sent.
+ * @return An #az_result value indicating the result of the operation.
+ */
+AZ_NODISCARD az_result
+az_hfsm_pipeline_send_inbound_event(az_hfsm_policy* policy, az_hfsm_event const event);
+
+/**
+ * @brief Sends an outbound event from the current policy.
+ *
+ * @note This function should be used during pipeline processing. The event is processed on the
+ * current thread (stack).
+ *
+ * @param policy The current policy trying to send the event.
+ * @param event The event being sent.
+ * @return An #az_result value indicating the result of the operation.
+ */
+AZ_NODISCARD az_result
+az_hfsm_pipeline_send_outbound_event(az_hfsm_policy* policy, az_hfsm_event const event);
+
+#ifdef TRANSPORT_MQTT_SYNC
+enum az_hfsm_event_type_pipeline
+{
+  /**
+   * @brief The event type posted by #az_hfsm_pipeline_sync_process_loop to allow syncrhonous
+   * pipeline event processing.
+   *
+   */
+  AZ_HFSM_PIPELINE_EVENT_PROCESS_LOOP = _az_HFSM_MAKE_EVENT(_az_FACILITY_IOT_MQTT, 9),
+};
+
+/**
+ * @brief Blocking processing loop.
+ * @note Application should call this only when TRANSPORT_MQTT_SYNC is defined.
+ * @details This call will allow the MQTT stack to perform synchonous I/O. The current thread is
+ * blocked until I/O is complete.
+ *
+ * @param pipeline The #az_hfsm_pipeline.
+ * @return The #az_result error code.
+ */
+AZ_NODISCARD az_result az_hfsm_pipeline_sync_process_loop(az_hfsm_pipeline* pipeline);
+#endif // TRANSPORT_MQTT_SYNC
+
+/**
+ * @brief Pipeline interval timer interface
+ *
+ */
+
+typedef struct
+{
+  az_platform_timer platform_timer;
+
+  struct
+  {
+    az_hfsm_pipeline* pipeline;
+  } _internal;
+} az_hfsm_pipeline_timer;
+
+/**
+ * @brief Creates an #az_platform_timer associated with an #az_hfsm_pipeline.
+ * @details When the timer elapses, a TIMEOUT _outbound_ message will be generated. The event.#data
+ *          contains a pointer to the original #az_platform_timer.
+ * @param pipeline The pipeline.
+ * @param[in, out] out_timer The populated timer structure.
+ * @return An #az_result value indicating the result of the operation.
+ */
+AZ_NODISCARD az_result
+az_hfsm_pipeline_timer_create(az_hfsm_pipeline* pipeline, az_hfsm_pipeline_timer* out_timer);
+
+#include <azure/core/_az_cfg_suffix.h>
+
+#endif //_az_HFSM_PIPELINE_H
```

Comment:
GH will flag missing NL at EOF - please make sure all files have it. Check for warnings in the PR.